src/epics/pages.js

```diff
@@ -10,51 +11,53 @@ const buildUrl = ( domain, user, pageid ) => {
+		// Get a list of  all of the unique users.
 		.flatMap( () => {
-			const users = store.getState().pages.reduce( ( set, page ) => {
+			const data = store.getState().pages.reduce( ( set, page ) => {
+				// Get all the users.
 				return set.concat( page.editors.keySeq() );
-			}, new Set() ).toArray();
-
-			return Observable.from( users );
-		} )
-		// Get all of the pages for each user.
-		.map( ( user ) => {
-			const pages = store.getState().pages.filter( ( page ) => !page.editors.has( user ) );
-
-			return { user, pages };
-		} )
-		// If the user is on all pages, there are no pages to deteremine.
-		.filter( ( data ) => !!data.pages.size )
-		.flatMap( ( data ) => {
-			const requests = data.pages.map( ( page ) => {
-				return Observable.ajax( {
-					url: buildUrl( store.getState().wikis.get( store.getState().query.wiki ).domain, data.user, page.id ),
-					crossDomain: true,
-					responseType: 'json'
+			}, new Set() )
+				.map( ( user ) => {
+					const pages = store.getState().pages.filter( ( page ) => !page.editors.has( user ) );
+					return { user, pages };
 				} )
-					.flatMap( ( ajaxResponse ) => {
-						if ( ajaxResponse.response.query.pages.length === 0 ) {
-							return page;
-						}
+				.filter( data => !!data.pages.size );
 
-						const item = ajaxResponse.response.query.pages[ 0 ];
+			// If there are no pages to retrieve, then we are done.
+			if ( !data.size ) {
+				return Observable.of( RevisionsActions.setStatusReady() );
+			}
 
-						return Observable.of( page.setIn( [ 'editors', data.user ], typeof item.revisions !== 'undefined' && item.revisions.length > 0 ) );
-					} );
-			} ).toArray();
+			return Observable.from( data.toArray() )
+				.flatMap( ( data ) => {
+					const requests = data.pages.map( ( page ) => {
+						return Observable.ajax( {
+							url: buildUrl( store.getState().wikis.get( store.getState().query.wiki ).domain, data.user, page.id ),
+							crossDomain: true,
+							responseType: 'json'
+						} )
+							.flatMap( ( ajaxResponse ) => {
+								if ( ajaxResponse.response.query.pages.length === 0 ) {
+									return page;
+								}
+								const item = ajaxResponse.response.query.pages[ 0 ];
+								return Observable.of( page.setIn( [ 'editors', data.user ], typeof item.revisions !== 'undefined' && item.revisions.length > 0 ) );
+							} );
+					} ).toArray();
 
-			return Observable.forkJoin( requests );
-		} )
-		.flatMap( ( data ) => {
-			const pages = data
-				.reduce( ( map, page ) => {
-					return map.set( page.id, page );
-				}, new Map() );
+					return Observable.forkJoin( requests );
+				} )
+				.flatMap( ( data ) => {
+					const pages = data
+						.reduce( ( map, page ) => {
+							return map.set( page.id, page );
+						}, new Map() );
 
-			return Observable.of( PagesActions.updatePages( pages ) );
+					return Observable.concat(
+						Observable.of( PagesActions.updatePages( pages ) ),
+						Observable.of( RevisionsActions.setStatusReady() )
```

Comment:
You can either remove the explicitly status ready, or remove `PAGES_UPDATE` from `src/reducers/revisions/status.js`. I think I would prefer for the former, but I'm fine either way if you have a good reason for the explicit action here.

Reply:
Good catch. Forgot to remove 'PAGES_UPDATE' from the reducer. I prefer calling the explicit action.